ec_ms_quantification/EC-MS_ixdat_tutorial.ipynb

```diff
@@ -0,0 +1,1017 @@
+    "    mol_lists=[\n",
+    "        [\"H2\"],  # left y-axis\n",
+    "        [\"O2\"],  # right\n",
+    "    ],\n",
+    "    tspan_bg=[\n",
+    "        None,\n",
+    "        [0, 20],\n",
+    "    ],  # [left, right] y-axes\n",
+    "    unit=[\"nmol/s\", \"pmol/s/cm^2\"],  # [left, right] y-axes\n",
+    "    logplot=False,\n",
+    "    legend=False, # e.g. if you want to add labels to the figure manually later\n",
+    ")\n",
+    "axes_a[0].set_ylabel(\"cal. sig. / [nmol s$^{-1}$]\")\n",
+    "axes_a[2].set_ylabel(\"cal. sig. / [pmol s$^{-1}$cm$^{-2}$]\")\n",
+    "\n",
+    "fig_a = axes_a[0].get_figure()\n",
+    "fig_a.savefig(\"fig_a.png\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ef2e8faa",
+   "metadata": {},
+   "source": [
+    " --- Cyclic voltammetry MS plot of two of the cycles ---"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "8d56ff17",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "meas_b = good_bit.as_cv() #by defining it as cv, meas_b.plot() would automatically plot vs potential\n",
+    "#as soon as it's CV you can index by \"cycle\"\n",
+    "meas_b.redefine_cycle(start_potential=0.4, redox=1)\n",
+    "#now you can choose where a cycle starts, here defined as potential = 0.4, anodic/oxidizing = 1 (or True), cathodic/reducing = 0 (or False)\n",
+    "#probably the syntax is inspired by biologic data, but it actually has nothing to do with the old biologic data\n",
+    "#initial cycle is 0, first time the set condition is met, starts cycle 1 etc.\n",
+    "\n",
+    "axes_b = meas_b[1].plot_vs_potential(\n",
+    "    mol_list=[\"H2\", \"O2\"],\n",
+    "    unit=\"pmol/s/cm^2\",\n",
+    "    logplot=False,\n",
+    "    legend=False,\n",
+    "    remove_background=True, #should remove the minimum value from the cut dataset, but it is possible somewhere to actually \n",
+    "    # the background (possibly where you define the dataset)\n",
+    ")\n",
+    "meas_b[3].plot_vs_potential(\n",
+    "    mol_list=[\"H2\", \"O2\"],\n",
+    "    unit=\"pmol/s/cm^2\",\n",
+    "    logplot=False,\n",
+    "    legend=False,\n",
+    "    remove_background=True,\n",
+    "    axes=axes_b, #to reuse the axis for co-plotting two separate parts of the measurement\n",
+    "    linestyle=\"--\",\n",
+    ")\n",
+    "axes_b[0].set_ylabel(\"cal. sig. / [pmol s$^{-1}$cm$^{-2}$]\")\n",
+    "axes_b[0].set_xlabel(meas_b.U_name)\n",
+    "\n",
+    "fig_b = axes_b[0].get_figure()\n",
+    "fig_b.savefig(\"fig_b.png\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "7d96e728",
+   "metadata": {},
+   "source": [
+    "EC-MS plot with carrier gases on right y-axis (uncalibrated) and products on left (calibrated)\n",
+    "-----------------------------------------------------------------------------------------------------------------------------------\n",
+    "Note that it is not possible to mix linear and logarithmic scales in one plot with ixdat."
```

Comment:
I think the main point here is that it *is* possible to mix calibrated and non-calibrated data, although the user might need an explanation of why they are being shown this, like:

"Note that it is possible to mix calibrated and uncalibrated data by calling the plotting function twice. This enables one to plot the uncalibrated carrier gas signal (which may carry information about how the experiment was done even if it is not itself quantified) on one y-axis and the product signal on the other y-axis."

And it *is* possible to have one of the y-axes be logplot and one linear. Also, a better unit could be used and I think left and right are switched in the comments. Here all that is corrected:
```
axes_c = meas_c.plot_measurement(
    mol_lists = [[], ["H2", "O2"]], # right y-axis
    unit="pmol/s",
    logplot=False,
    legend=False,
)
meas_c.plot_measurement(
    mass_list=["M4", "M28", "M40"],  # left y-axis
    logplot=True,
    legend=False,
    axes=axes_c, #to reuse the axis for co-plotting two separate parts of the measurement
    #linestyle="--",
)
axes_c[2].set_ylabel("cal. sig. / [nmol s$^{-1}$]")
# axes_c[-1].set_ylim(bottom=-5)
```

Reply:
I was using an outdated version of Jupyter (and matplotlib) when I first made the tutorial and I'm slowly realising that a number of the limitations I'm suggesting workarounds for are due to that..